aspnetcore/blazor/file-uploads.md

```diff
@@ -74,21 +77,295 @@ The following example demonstrates multiple image file upload in a component. `I
                 100, 100);
             var buffer = new byte[resizedImageFile.Size];
             await resizedImageFile.OpenReadStream().ReadAsync(buffer);
-            var imageDataUrl = 
+            var imageData = 
                 $"data:{format};base64,{Convert.ToBase64String(buffer)}";
-            imageDataUrls.Add(imageDataUrl);
+            images.Add(imageData);
         }
     }
 }
 ```
 
 `IBrowserFile` returns metadata [exposed by the browser](https://developer.mozilla.org/docs/Web/API/File#Instance_properties) as properties. This metadata can be useful to preliminary validation. For example, see the [`FileUpload.razor` and `FilePreview.razor` sample components](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/blazor/file-uploads/samples/).
 
+## Upload files to a server
+
+**Content is TBD based on firming up the example code.**
+
+**Include STRONG security warning with cross-link to security guidance in https://docs.microsoft.com/aspnet/core/mvc/models/file-uploads.**
+
+`UploadResult.cs` in the **`Shared`** project of a hosted Blazor WebAssembly solution:
+
+```csharp
+public class UploadResult
+{
+    public bool Uploaded { get; set; }
+
+    public string FileName { get; set; }
+
+    public string StoredFileName { get; set; }
+
+    public int ErrorCode { get; set; }
+}
+```
+
+`Pages/UploadFiles.razor` in the **`Client`** project of a hosted Blazor solution:
+
+```razor
+@page "/upload-files"
+@using System.IO
+@using System.Linq
+@using System.Net
+@using Microsoft.Extensions.Logging
+@using FileUploadTesting.Shared
+@inject HttpClient Http
+@inject ILogger<UploadFiles> logger
+
+<h1>Upload files</h1>
+
+<p>
+    Upload one or more PNG image files:
+    <InputFile OnChange="@OnInputFileChange" multiple />
+</p>
+
+@if (images.Count > 0)
+{
+    <div class="card">
+        <div class="card-body">
+            @foreach (var image in images)
+            {
+                <figure>
+                    <img class="rounded m-1" src="@image.Thumbnail" />
+                    <figcaption>
+                        @WebUtility.HtmlEncode(image.Name)
```

Comment:
Razor already HTML encodes everything it emits via an `@`. This would get double encoded.

Reply:
Ah, yes. I recall confronting that concept now.